WordPress/gutenberg
# Patterns: update the Manage all my patterns command to redirect to site editor patterns list

## What?
Updates the `Manage all my patterns` command to redirect the site editor patterns page when using a block-based theme.

## Why?
Fixes: #52491

## How?
Checks for block based theme, or site editor context and redirects accordingly.

## Testing Instructions

- In  a block-based theme use `Manage all my patterns` command in site editor and post editor and make sure it redirects to the site editor patterns page
- In a non-block theme use `Manage all my patterns` command in  post editor make sure it redirects to the wp-admin patterns management page
- In a block-based theme, when logged in as an `author` level user check that the command in the post editor redirects to   wp-admin patterns page and not the site editor

## Screenshots or screencast 

Block-based theme:

https://github.com/WordPress/gutenberg/assets/3629020/454dec22-47d6-4f93-923a-3854578fa5e6

Non block theme:

https://github.com/WordPress/gutenberg/assets/3629020/e75c0db0-2463-46f5-9b4a-1236a4d52afc
